hw_diag/views/diagnostics.py

```diff
@@ -68,52 +70,21 @@ def get_initialisation_file():
     so we bypass the regular timer.
     """
 
-    diagnostics = {}
-    get_rpi_serial(diagnostics)
-    get_ethernet_addresses(diagnostics)
-    get_environment_var(diagnostics)
-    set_diagnostics_bt_lte(diagnostics)
-    ecc_tests = get_gateway_mfr_test_result()
-    public_keys = get_public_keys_rust()
-
-    if ecc_tests['result'] == 'pass':
-        diagnostics["ECC"] = True
-    else:
-        return 'ECC tests failed', 503
-
-    if lora_module_test():
-        diagnostics["LOR"] = True
-    else:
-        return 'LoRa Module is not ready', 503
-
-    if (
-            diagnostics["ECC"]
-            and diagnostics["E0"]
-            and diagnostics["W0"]
-            and diagnostics["BT"]
-            and diagnostics["LOR"]
-    ):
-        diagnostics["PF"] = True
-    else:
-        diagnostics["PF"] = False
-
-    try:
-        diagnostics['OK'] = public_keys['key']
-        diagnostics['PK'] = public_keys['key']
-    except KeyError:
-        return 'Internal Server Error', 500
-
-    response = {
-        "VA": diagnostics['VA'],
-        "FR": diagnostics['FR'],
-        "E0": diagnostics['E0'],
-        "W0": diagnostics['W0'],
-        "RPI": diagnostics['RPI'],
-        "OK": diagnostics['OK'],
-        "PK": diagnostics['PK'],
-        "PF": diagnostics["PF"],
-        "ID": diagnostics["ID"]
-    }
-
+    diagnostics = [
+        RpiSerialDiagnostic(),
+        EccDiagnostic(),
+        MacDiagnostics(),
+        EnvVarDiagnostics(),
+        BtLteDiagnostics(),
+        LoraDiagnostic(),
+        KeyDiagnostics(),
+        # Must be last, it depends on previous results
+        PfDiagnostic()
+    ]
+    diagnostics_report = DiagnosticsReport(diagnostics)
+    diagnostics_report.perform_diagnostics()
+
+    keys_to_extract = ["VA", "FR", "E0", "W0", "RPI", "OK", "PK", "PF", "ID"]
+    response = diagnostics_report.get_report_subset(keys_to_extract)
```

Comment:
Is there harm in returning the full report here? I'd like to include at least the relevant friendly keys too. Ideally, the full report.